pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/partition/Murmur3PartitionFunction.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.spi.partition;
+
+import com.google.common.base.Preconditions;
+import com.google.common.hash.Hashing;
+import java.util.Map;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+
+/**
+ * Implementation of {@link PartitionFunction} which partitions based on 32 bit murmur3 hash
+ */
+public class Murmur3PartitionFunction implements PartitionFunction {
+  private static final String NAME = "Murmur3";
```

Comment:
interestingly we already have MURMUR3 in https://github.com/apache/pinot/blob/b9ed378355acc5313ebc1031510671f7045f29a9/pinot-spi/src/main/java/org/apache/pinot/spi/config/table/HashFunction.java#L4

Reply:
+1. Also Guava already provides the murmur3 implementation (used in `HashUtils`), do we need to copy it from another library?

Reply:
We want to add the support murmur3 for debezium. Unfortunately, Debezium is using a special variant x64_32 that is not available for many library that we looked into (the one `HashFunction` as based on as well as the one from apache commons). I think that @aishikbh had to copy for `x64_32` variant.

Reply:
:+1: